src/components/DeleteItem.jsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from "react"
+import { deleteItem } from "../utils/api-requests"
+import Loading from "./Loading"
+import ErrorHandling from "./ErrorHandling"
+
+function DeleteItem({id, articlesOrComments, setCommentsChanged, setArticlesChanged}) {
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState(null)
+
+  
+  function handleClick(){
+    setIsLoading(true)
+    deleteItem(id, articlesOrComments).then(()=>{
```

Comment:
Really happy to see you're triggering this on the click and not a useEffect (which is not necessary in this instance). This does mean you can delete the import of useEffect on line 1 though